client/account-settings/index.js

```diff
@@ -32,6 +32,15 @@ export default ( { formData, formMeta, storeOptions } ) => ( {
 		};
 	},
 
+	getStateForPersisting( state ) {
+		delete state.notices;
```

Comment:
Don't mutate the `state` variable, return a new state tree instead. Since this is run in the `onbeforeunload` handler, if the user chooses to "stay", the `notices` will be gone from the Redux state without Redux even noticing to trigger a re-render.